modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java

```diff
@@ -2629,10 +2661,53 @@ public boolean isAutoActivationEnabled() {
      * @param autoActivation Auto activation enabled flag value.
      * @return {@code this} instance.
      * @see #isAutoActivationEnabled()
+     * @deprecated Use {@link IgniteConfiguration#setClusterStateOnStart(ClusterState)} instead.
      */
+    @Deprecated
     public IgniteConfiguration setAutoActivationEnabled(boolean autoActivation) {
+        U.warn(log, "Property autoActivation deprecated. Use clusterStateOnStart instead.");
+
         this.autoActivation = autoActivation;
 
+        autoActivationPropSetFlag = true;
+
+        return this;
+    }
+
+    /**
+     * Gets state of cluster on start.
+     * <br/>
+     * For <b>in-memory cluster</b> this state will be applied to the first started node. If
+     * cluster state on start is {@link ClusterState#INACTIVE}, there will be no cache partition map exchanges
```

Comment:
What is partition map exchange?
Let's explain behavior to user in more understandable terms.
I'd describe impact here:
1) Further node joins will be handled by cluster faster
2) Manual activation should be performed in order to start working with the cluster

Reply:
done